src/realm/sync/subscriptions.hpp

```diff
@@ -278,13 +289,30 @@ class SubscriptionStore {
         ColKey subscriptions;
     };
 
+    struct NotificationRequest {
+        NotificationRequest(int64_t version, util::Promise<void> promise, SubscriptionSet::State notify_when)
```

Comment:
```suggestion
        NotificationRequest(int64_t version, util::Promise<void>&& promise, SubscriptionSet::State notify_when)
```
Seems like this ctor is only called with promise as an rvalue ref.

Reply:
Promises are move-only types, no need to qualify them that way.